packages/coingecko/test/helpers.test.ts

```diff
@@ -0,0 +1,31 @@
+import { expect } from 'chai'
+
+import { getCoingeckoSimplePriceUri, getCoingeckoPriceFetch } from '../src/helpers'
+
+describe('getCoingeckoSimplePrice', () => {
+  it('ethereum in usd price', () => {
+    const url = 'https://api.coingecko.com/api/v3/simple/price?ids=ethereum&vs_currencies=usd'
+    expect(getCoingeckoSimplePriceUri('ethereum', 'usd')).to.eq(url)
+  })
+
+  it('bitcoin in usd price', () => {
+    const url = 'https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd'
+    expect(getCoingeckoSimplePriceUri('bitcoin', 'usd')).to.eq(url)
+  })
+
+  it('dai in usd price', () => {
+    const url = 'https://api.coingecko.com/api/v3/simple/price?ids=dai&vs_currencies=usd'
+    expect(getCoingeckoSimplePriceUri('dai', 'usd')).to.eq(url)
+  })
+})
+
+describe('getCoingeckoPriceFetch', () => {
+  it('Success', async () => {
+    const mockFetch = () =>
+      Promise.resolve({
+        json: () => ({ ethereum: { usd: 2234.6 } }),
+      })
+    const getCoingeckoPrice = getCoingeckoPriceFetch(mockFetch)
+    expect(await getCoingeckoPrice('ethereum', 'usd')).to.eq(2234.6)
+  })
```

Comment:
Also please add tests for `no answer` and `wrong answer`

```ts
  it('no answer', async () => {
    const mockFetch = () => throw()
    const getCoingeckoPrice = getCoingeckoPriceFetch(mockFetch)
    expect(await getCoingeckoPrice('ethereum', 'usd')).to.eq(undefined)
  })
```

```ts
  it('wrong answer', async () => {
    const mockFetch = () =>
      Promise.resolve({
        json: () => ({ eth: { usd: 2234.6 } }),
      })
    const getCoingeckoPrice = getCoingeckoPriceFetch(mockFetch)
    expect(await getCoingeckoPrice('ethereum', 'usd')).to.eq(undefined)
  })
```

Reply:
@Szymx95 we currently handle the `setPrice(undefined)` in `useCoingeckoPrice` instead of `getCoingeckoPriceFetch`,  would you think its better to move the try..catch handle in `getCoingeckoPriceFetch` so we can add the above tests?

Reply:
let's just leave the `getCoingeckoPriceFetch` as is. And in tests just test if the function throws